indigo/indigo/src/main/scala/indigo/shared/formats/GenericTileset.scala

```diff
@@ -0,0 +1,39 @@
+package indigo.shared.formats
+
+import indigo._
+
+/** Utility for working with generic sprite sheets. The
+  * sheet must be uniform, with each sprite having the same
+  * width and height, and a (possibly zero-pixel) margin.
+  */
+final case class GenericTileset(
+    assetName: AssetName,
+    width: Int,
+    height: Int,
+    tileWidth: Int,
+    tileHeight: Int,
```

Comment:
The convention is that the main constructor will use `Size` instead of separate width and height values. What we then do is has a companion with an apply method that gives additional (more loosely typed) construction options. So this would be (perhaps):

```scala
assetName: AssetName, imageSize: Size, tileSize: Size, margin: Int
```

Reply:
So, move the current argument list to an apply method? OK.